engine/src/test/java/io/zeebe/engine/job/CompleteJobTest.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Zeebe Workflow Engine
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.zeebe.engine.job;
+
+import static io.zeebe.test.util.record.RecordingExporter.jobRecords;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.ThrowableAssert.catchThrowable;
+
+import io.zeebe.engine.processor.workflow.MsgPackConstants;
+import io.zeebe.engine.util.EngineRule;
+import io.zeebe.exporter.api.record.Assertions;
+import io.zeebe.exporter.api.record.Record;
+import io.zeebe.exporter.api.record.RecordMetadata;
+import io.zeebe.exporter.api.record.value.JobBatchRecordValue;
+import io.zeebe.exporter.api.record.value.JobRecordValue;
+import io.zeebe.msgpack.spec.MsgPackHelper;
+import io.zeebe.protocol.RecordType;
+import io.zeebe.protocol.RejectionType;
+import io.zeebe.protocol.impl.encoding.MsgPackConverter;
+import io.zeebe.protocol.intent.JobIntent;
+import io.zeebe.test.util.MsgPackUtil;
+import io.zeebe.test.util.record.RecordingExporter;
+import org.agrona.DirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class CompleteJobTest {
+  private static final String JSON_VARIABLES = "{\"foo\":\"bar\"}";
+  private static final byte[] VARIABLES_MSG_PACK = MsgPackUtil.asMsgPackReturnArray(JSON_VARIABLES);
+  private static final String JOB_TYPE = "foo";
+  private static final String PROCESS_ID = "process";
+
+  @Rule public EngineRule engineRule = new EngineRule();
+
+  @Test
+  public void shouldCompleteJob() {
+    // given
+    engineRule.createJob(JOB_TYPE, PROCESS_ID);
+    final Record<JobBatchRecordValue> batchRecord =
+        engineRule.jobs().withType(JOB_TYPE).activateAndWait();
```

Comment:
In some tests I don't want to wait for the ACTIVATED or COMPLETED record because the command is supposed to be rejected. That's the only reason why I separated the writing of the command and the waiting for the record. In that case, I'll change this into an `.await()` method that waits for the follow-up event so we avoid having multiple things happening in the method with a conjunction. Do you agree?

Reply:
Ok got it. This is the reason why I added on some clients the method `expectRejection()`, which then waits until the rejeciton published instead of the follow up event. https://github.com/zeebe-io/zeebe/blob/develop/engine/src/test/java/io/zeebe/engine/util/PublishMessageClient.java#L99 or https://github.com/zeebe-io/zeebe/blob/74129fa41b14f645a01d59bfc357c963a98890b7/engine/src/test/java/io/zeebe/engine/util/DeploymentClient.java#L120

Reply:
Ah I hadn't noticed that, that's a better solution. Thanks!